scenarios/pubsub_simple.erl

```diff
@@ -0,0 +1,182 @@
+-module(pubsub_simple).
+
+-behavior(amoc_scenario).
+
+-include_lib("exml/include/exml.hrl").
+-include_lib("escalus/include/escalus.hrl").
+
+-define(HOST, <<"localhost">>). %% The virtual host served by the server
+-define(SERVER_IPS, {<<"127.0.0.1">>}). %% Tuple of servers, for example {<<"10.100.0.21">>, <<"10.100.0.22">>}
+-define(PUBSUB_ADDR, <<"pubsub.localhost">>).
+-define(WAIT_FOR_STANZA_TIMEOUT, 10000).
+-define(GROUPNAME, <<"pubsub_test_group_name">>).
+-define(NUMBER_OF_PUBSUB_NODES, 5).
+-define(DELAY_BETWEEN_MESSAGES, 100).
+-define(WAIT_FOR_NODES, 5000).
+
+-export([init/0]).
+-export([start/1]).
+
+-spec init() -> ok.
+init() -> ok.
+
+-spec start(integer()) -> ok.
+
+start(1) ->
+    pg2:create(?GROUPNAME),
+    pg2:join(?GROUPNAME, self()),
+    Client = connect_amoc_user(1),
+    Nodes = [create_pubsub_node(Client) || _ <- lists:seq(1, ?NUMBER_OF_PUBSUB_NODES)],
+    provide_nodes_names(Client, Nodes);
+
+start(Id) ->
+    pg2:create(?GROUPNAME),
+    Client = connect_amoc_user(Id),
+    Nodes = get_nodes(3, Client),
+    case Id rem 2 of
+        0 ->
+            lager:info("Client: Starting publisher\n", []),
+            publish_items_forever(Client, Id, Nodes, 1);
+        1 ->
+            lager:info("Client: Starting subscriber\n", []),
+            work_as_subscriber(Id, Client, Nodes)
+    end,
+    ok.
+
+provide_nodes_names(SelfClient, Nodes) ->
+    receive
+        {what_nodes, PID, Client} ->
+            AffChange = [{Client, <<"publisher">>}],
+            [ set_affiliations(SelfClient, Node, AffChange) || Node <- Nodes],
```

Comment:
As far as I remember, the nodes can be configured (while creating) in such a way that publishers can join by themselves, am I right?

Reply:
right